dolphinscheduler-master/src/test/java/org/apache/dolphinscheduler/server/master/integration/cases/WorkflowStartTestCase.java

```diff
@@ -121,7 +159,65 @@ public void testStartWorkflow_with_subWorkflowTask_success() {
                             .matches(
                                     workflowInstance -> workflowInstance.getState() == WorkflowExecutionStatus.SUCCESS)
                             .matches(
-                                    workflowInstance -> workflowInstance.getIsSubWorkflow() == Flag.NO);
+                                    workflowInstance -> workflowInstance.getIsSubWorkflow() == Flag.NO)
+                            .matches(
+                                    workflowInstance -> workflowInstance.getDryRun() == Flag.NO.getCode());
+
+                    final List<WorkflowInstance> subWorkflowInstance =
+                            repository.queryWorkflowInstance(context.getWorkflows().get(1));
+                    Assertions
+                            .assertThat(subWorkflowInstance)
+                            .hasSize(1)
+                            .satisfiesExactly(workflowInstance -> {
+                                assertThat(workflowInstance.getState()).isEqualTo(WorkflowExecutionStatus.SUCCESS);
+                                assertThat(workflowInstance.getIsSubWorkflow()).isEqualTo(Flag.YES);
+                                assertThat(workflowInstance.getDryRun()).isEqualTo(Flag.NO.getCode());
+                            });
+
+                    Assertions
+                            .assertThat(repository.queryTaskInstance(workflowInstanceId))
+                            .satisfiesExactly(taskInstance -> {
+                                assertThat(taskInstance.getName()).isEqualTo("sub_logic_task");
+                                assertThat(taskInstance.getState()).isEqualTo(TaskExecutionStatus.SUCCESS);
+                            });
+
+                    Assertions
+                            .assertThat(repository.queryTaskInstance(subWorkflowInstance.get(0).getId()))
+                            .satisfiesExactly(taskInstance -> {
+                                assertThat(taskInstance.getName()).isEqualTo("fake_task");
+                                assertThat(taskInstance.getState()).isEqualTo(TaskExecutionStatus.SUCCESS);
+                            });
+                });
+
+        assertThat(workflowRepository.getAll()).isEmpty();
+    }
+
+    @Test
+    @DisplayName("Test start a workflow with one sub workflow task(A) dry run success")
+    public void testStartWorkflow_with_subWorkflowTask_dryRunSuccess() {
```

Comment:
Why the parent workflow using dry run but still generate sub workflow instance? If use dry run then the SubWorkflow task will not generate sub workflow instance.

Reply:
> Why the parent workflow using dry run but still generate sub workflow instance? If use dry run then the SubWorkflow task will not generate sub workflow instance.

Child workflow inherits the dry run configuration of the parent workflow.
This is indeed the effect, and the sub-workflow will also dry run.

<img width="831" alt="image" src="https://github.com/user-attachments/assets/6a986bb9-e6c3-4ccd-aed8-b953a585cfdb">

<img width="1661" alt="image" src="https://github.com/user-attachments/assets/bcdebc15-2ad4-4c3d-ba78-444fe5e0c5f8">

Reply:
This might make confusion, since other task dry run will not go into the task plugin logic. It's better to keep this consistent.

You should do this change in `MasterTaskExecutor`, otherwise the other logic task still doesn't support dry run.

Reply:
> This might make confusion, since physical task dry run will not go into the task plugin logic, you can find the implementation in `WorkerTaskExecutor`. It's better to keep this consistent.
> 
> You should do this change in `MasterTaskExecutor`, otherwise the other logic task still doesn't support dry run.

OK, I'll check it out.

Reply:
> This might make confusion, since physical task dry run will not go into the task plugin logic, you can find the implementation in `WorkerTaskExecutor`. It's better to keep this consistent.
> 
> You should do this change in `MasterTaskExecutor`, otherwise the other logic task still doesn't support dry run.

Check whether it is dry run in AsyncMasterTaskDelayQueueLooper, it may also be necessary!